### PR TITLE
passport: add select level requestor

### PR DESCRIPTION
**Pull request by walkawayy**
In this commit, the requestor starts a new game at the selected
level. In a future PR, the selected level will start with the
starting items from the selected savegame's playthrough.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description
passport: add select level requestor
    
In this commit, the requestor starts a new game at the selected level. In a future PR, the selected level will start with the starting items from the selected savegame's playthrough.
...

### CODE REVIEW

**Review by rr- — COMMENTED** [review]
LGTM apart from a few cosmetic remarks.

File: src/game/option_passport.c [hunk 1]
```diff
@@ -156,12 +229,55 @@ static void Option_PassportInitNewGameRequester(void)
...
+        req->vis_lines = 10;
+    } else {
+        req->y = -120;
+        req->vis_lines = 12;
+    }
+
+    Savegame_ScanAvailableLevels(req);
+}
+
+static void Option_PassportInitSaveRequester(int16_t page_num)
```
[the remark this answers is not in the document]
Comment (walkawayy) [reply src/game/option_passport.c:271]: These are under the `REQUEST_INFO` requestor declarations. I don't see a missing one.
```
static void Option_PassportInitText(void);
static void Option_PassportShutdownText(void);
static void Option_PassportShowNewGame(void);
static void Option_PassportShowSaves(void);
static void Option_PassportInitLevelStrings(void);
static void Option_PassportFreeLevelStrings(void);
static void Option_PassportInitText(void);
static void Option_PassportInitNewGameRequester(void);
static void Option_PassportInitSaveRequester(int16_t page_num);
static void Option_PassportLoadGame(void);
static void Option_PassportSelectLevel(void);
```

File: src/game/option_passport.c [hunk 2]
```diff
@@ -156,12 +229,55 @@ static void Option_PassportInitNewGameRequester(void)
...
+{
+    REQUEST_INFO *req = &g_LoadSavegameRequester;
+    InitRequester(req);
+    SetRequesterHeading(
+        req,
+        page_num == PASSPORT_PAGE_1
+            ? g_GameFlow.strings[GS_PASSPORT_LOAD_GAME]
+            : g_GameFlow.strings[GS_PASSPORT_SAVE_GAME]);
```
[the remark this answers is not in the document]
Comment (walkawayy) [reply src/game/option_passport.c:279]: I'm not sure if I'm missing something obvious but idk what you mean on this comment. Inside the `g_GameFlow.strings[GS_PASSPORT_LOAD_GAME]` brackets?

[the remark this answers is not in the document]
Comment (rr-) [reply src/game/option_passport.c:279]: `g_GameFlow.strings[... ? GS_... : GS_...]`

**Review by rr- — APPROVED** [review]
LGTM. What do you think about shortening m_LoadSavegameStrings to m_LoadGameStrings and g_LoadSavegameRequester to g_LoadGameRequester?

**Comment by walkawayy** [comment]
> LGTM. What do you think about shortening m_LoadSavegameStrings to m_LoadGameStrings and g_LoadSavegameRequester to g_LoadGameRequester?

I don't think your proposed names would make sense tbh. But I think `g_SavegameRequester` and `m_SavegameStrings ` would.

**Comment by rr-** [comment]
@walkawayy The idea was to not merge half-finished feature to master and open a PR based on this PR

**Comment by walkawayy** [comment]
Oof well we can revert it. I was mixed up because of the stable branch talk.

**Comment by rr-** [comment]
Let's keep it to avoid more bureaucracy but for the future let's try to avoid this.

**Pull request merged**